python/kvikio/kvikio/_lib/libnvcomp_ll.pyx

```diff
@@ -371,20 +372,26 @@ class nvCompBatchAlgorithmLZ4(nvCompBatchAlgorithm):
 
     HEADER_SIZE_BYTES: size_t = sizeof(uint32_t)
 
-    def __init__(self, data_type: int = 0, has_header: bool = True):
+    def __init__(self, data_type: int = None, has_header: bool = True):
         """Initialize the codec.
 
         Parameters
         ----------
-        data_type: int
-            Source data type.
+        data_type: int or None
+            Source data type. If None, uses nvcomp default options.
         has_header: bool
             Whether the compressed data has a header.
             This enables data compatibility between numcodecs LZ4 codec,
             which has the header and nvCOMP LZ4 codec which does not
             require the header.
         """
-        self.options = nvcompBatchedLZ4Opts_t(data_type)
+        if data_type is None:
+            self.options = nvcompBatchedLZ4DefaultOpts
+        elif isinstance(data_type, int):
+            self.options = nvcompBatchedLZ4Opts_t(data_type)
+        else:
+            raise ValueError(f"Invalid value for data_type: {data_type}")
```

Comment:
*nit (non-blocking)*: This is arguably a `TypeError`, not a `ValueError`. We've passed an incorrect type, not a correct type with an incorrect value.

Reply:
Agreed. Though Cython generates this check for us. So we can drop it altogether

There's an example in the top review comment

https://github.com/rapidsai/kvikio/pull/450#pullrequestreview-2271498081